modules/ROOT/pages/configure-multiple-credential-providers.adoc

```diff
@@ -0,0 +1,116 @@
+= Configuring Multiple Identity Providers for Client Management
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: credential providers, external client providers
+
+You can use multiple client providers, to help you enforce security and regulations in your business organization. These client providers, such as, OpenAM and PingFederate, enable you to secure your operational data, such as client credentials and access token. 
+
+You can use an Anypoint Platform client provider (native client provider) or implement an external client provider. If you do not specify a client provider for your environment or business group, Anypoint Platform uses its native client provider by default.
+
+To assign separate client providers for different organizations and environments, you must first enable clients applications to be authenticated using OAuth for the client providers that you want to implement in your organization.
+
+When you use multiple client providers, you can use them to:
+
+* Different business groups
+* Different environments, such as production, QA, and sandbox
+* Internal and external APIs
+
+== How This Works
+
+//Need help here
+
+When you enable external client providers, they act as authorization servers to issue access tokens to a client application for accessing an API. When you request access to a client application, the corresponding client ID and client secret credentials are created in the IdP specified in that API specification.
+
+If an API has only applied the Client ID Enforcement policy and not applied the OAuth policy, the client application is still created in an external IdP.
+When an OAuth policy is deployed to API gateway, the platform automatically injects IdP configuration, such as the token URL and token introspection url in the policy configuration. The IdP configuration is completely hidden from a user in API Manager.
+
+If no external IdPs are enabled, Access Manager acts as the default authorization server in that case. 
+//[PA:-Tomas not sure what the following statement is trying to say.]
+// OOTB OAuth policies with an external IdP can’t be applied 
+// ClientId enforcement policy can be applied.
+
```

Comment:
Answering the comment:
It means that if the customer doesn't associate an external client provider, there are a set of policies that they can't apply (those that rely on an external client provider)

Reply:
Ok that is already covered.